crates/block-producer/src/block_builder/mod.rs

```diff
@@ -149,14 +149,14 @@ impl BlockBuilder {
         let (block_header_witness, updated_accounts) = BlockWitness::new(block_inputs, batches)?;
 
         let new_block_header = self.block_kernel.prove(block_header_witness)?;
-        let block_num = new_block_header.block_num();
 
         // TODO: return an error?
         let block =
             Block::new(new_block_header, updated_accounts, output_notes, produced_nullifiers)
                 .expect("invalid block components");
 
         let block_hash = block.hash();
+        let block_num = new_block_header.block_num().as_u32();
```

Comment:
Does `BlockNumber` not implement `Display`? I think that's all this was used for or no?

Reply:
It does, updated the calls.